### ISSUE TITLE
Bug Report:  Expired/Invalid JWT-Token and CORS-errors / FISH-6299

### ISSUE THREAD

**Issue by UlrichCech** [payara/Payara#5746]
# Description #
----------

I have a JavaEE application, which runs just fine with Java 11 and Payara until 5.2021.5.

I think, with the Payara version 5.2021.6 (tested and verified in version 5.2022.2), the behavior of the JWT-handling changed in that way, that there is an exception throw, if the provided JWT-token is expired or invalid. This has the following implications:
The request is finished, before the request reaches any filter or interceptor of the application. And with that issue, any CORS headers are not set, which rises CORS-errors in the Frontend, which cannot handle this situation anymore.

In version 5.2021.5, there is no "fish.payara.microprofile.jwtauth.eesecurity.JWTProcessingException" and the token is propagated to the application, where I could handle this consistent and return a concise response to the Frontend.

In addition, that an expired token is a really "normal behavior" and occurs often, the server-log is polluted with this exception-stacktrace.

The administrative problem is, that we are stuck on the Payara 5.2021.5 version, and in some days, the new Payara 6 version comes out and the support of Payara 5 will be finished!

## Expected Outcome ##

There should be no exception thrown and the token should be propagated again through any filter and interceptors of the application in order to be handled accordingly (like it was in PayaraServer-5.2021.5).

## Current Outcome ##

This exception is thrown and there are no CORS-headers:

```
fish.payara.microprofile.jwtauth.eesecurity.JWTProcessingException: fish.payara.microprofile.jwtauth.eesecurity.JWTProcessingException: JWT token expired
	at fish.payara.microprofile.jwtauth.jwt.JwtTokenParser.parse(JwtTokenParser.java:152)
	at fish.payara.microprofile.jwtauth.eesecurity.SignedJWTIdentityStore.validate(SignedJWTIdentityStore.java:116)
	at javax.security.enterprise.identitystore.IdentityStore.validate(IdentityStore.java:85)
	at fish.payara.microprofile.jwtauth.eesecurity.SignedJWTIdentityStore$Proxy$_$$_WeldClientProxy.validate(Unknown Source)
	at org.glassfish.soteria.cdi.DefaultIdentityStoreHandler.validate(DefaultIdentityStoreHandler.java:96)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.weld.bean.proxy.AbstractBeanInstance.invoke(AbstractBeanInstance.java:38)
	at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:106)
	at org.jboss.weld.generated.proxiesx.security.enterprise.identitystore.IdentityStoreHandler$1763020431$Proxy$_$$_WeldClientProxy.validate(Unknown Source)
	at fish.payara.microprofile.jwtauth.eesecurity.JWTAuthenticationMechanism.validateRequest(JWTAuthenticationMechanism.java:100)
	at fish.payara.microprofile.jwtauth.eesecurity.JWTAuthenticationMechanism$Proxy$_$$_WeldClientProxy.validateRequest(Unknown Source)
	at org.glassfish.soteria.mechanisms.jaspic.HttpBridgeServerAuthModule.validateRequest(HttpBridgeServerAuthModule.java:151)
	at org.glassfish.soteria.mechanisms.jaspic.DefaultServerAuthContext.validateRequest(DefaultServerAuthContext.java:76)
	at com.sun.web.security.realmadapter.JaspicRealm.validateRequest(JaspicRealm.java:391)
	at com.sun.web.security.realmadapter.JaspicRealm.validateRequest(JaspicRealm.java:358)
	at com.sun.web.security.realmadapter.JaspicRealm.validateRequest(JaspicRealm.java:181)
	at com.sun.web.security.RealmAdapter.invokeAuthenticateDelegate(RealmAdapter.java:487)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:468)
	at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:726)
	at org.apache.catalina.core.StandardPipeline.doChainInvoke(StandardPipeline.java:581)
	at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:97)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:158)
	at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:371)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:238)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: fish.payara.microprofile.jwtauth.eesecurity.JWTProcessingException: JWT token expired
	at fish.payara.microprofile.jwtauth.jwt.JwtTokenParser.verifyAndParseSignedJWT(JwtTokenParser.java:192)
	at fish.payara.microprofile.jwtauth.jwt.JwtTokenParser.parse(JwtTokenParser.java:147)
	... 45 more
```

## Steps to reproduce ## 

1. checkout the linked "reproducer project"
https://github.com/UlrichCech/payara-jwt-reproducer/tree/Payara-5-2022.2_JWT-Token_and_CORS_problem
on branch "Payara-5-2022.2_JWT-Token_and_CORS_problem"

2. in the sourcecode-directory "doc", there is a prepared HTML-file with a Button and a Javascript to send an XHttpRequest to localhost:8080.

3. Start the Reproducer-App, load the prepared HTML-file from the doc-Directory, go to the Network-Console of any browser and klick the Button... 

You will see the OPTIONS request with a status code OK 200 and all CORS headers, and the following GET-request, which fails with CORS-error... and the Exception (see above) in the server.log

## Environment ##

- **Distribution**: Server Full Profile - Version 5.2021.5, 5.2022.2
- **JDK Version**: OpenJDK 11
- **Operating System**: Mac

**Comment by AlanRoth** [comment]
Hi @UlrichCech,

Thank you for opening a issue and sharing the reproducer, I have been able to reproduce your error.

5.2020.5 would correspond to a component upgrade for Nimbus JOSE+JWT to version 8.20 from 5.4, which might be related.

I have raised `FISH-6299`. If you feel confident to implement a fix yourself, we welcome you too - otherwise our development team will prioritize and work on the issue accordingly, which has no guaranteed timeline.

Thank you,
Alan
